src/cli/serve/compile_config_stack.test.js

```diff
@@ -74,6 +74,43 @@ describe('compileConfigStack', () => {
     ]);
   });
 
+  it.each(['search_ai_lake', 'essentials', 'complete'])(
```

Comment:
__optional nit:__ it might be worth adding a test for an empty `productTypes` array and one with multiple values to ensure Kibana doesn't choke and does the right thing in these cases.